research/02-repl.tex

```diff
@@ -1,5 +1,172 @@
 \section{Read-Eval-Print Loops}
-\label{sec:read-eval-print}
+\label{sec:repl}
+
+% TODO: mention that REPLs do not necessarily have to be interactive?
+Many programming languages come with an interactive environment. This
+interactive environment is an interface to the programming language's execution
+engine. These interactive environments know many different names, including but
+not limited to \textit{language shell}, \textit{command-line interpreter} and
+\textit{REPL}. There are subtle differences between these interpretations.
+These are, however, mostly of semantic value. In this report the term
+REPL is chosen, because it conveys the notion of such an interactive
+environment well.
+
+The Lisp family of programming languages is believed to be amongst the
+first programming languages offering such functionality~\cite{NEEDED}. The name
```

Comment:
I think a citation would indeed be nice. Although Lisp pioneered in many things, so maybe it's not that hard to imagine :P.

Maybe you can find a paper about the "history of REPLs", or about the "history of interactive environments"?